extmod/modtls_mbedtls.c

```diff
@@ -250,13 +252,30 @@ STATIC mp_obj_t ssl_context_make_new(const mp_obj_type_t *type_in, size_t n_args
 
     return MP_OBJ_FROM_PTR(self);
 }
+STATIC int ssl_sock_cert_verify(void *ptr, mbedtls_x509_crt *crt, int depth, uint32_t *flags) {
+    mp_obj_ssl_socket_t *o = ptr;
+    if (o->ssl_context->handler == mp_const_none) {
+        return 0;
+    }
+    return mp_obj_get_int(mp_call_function_2(o->ssl_context->handler, mp_obj_new_bytes(crt->raw.p, crt->raw.len), MP_OBJ_NEW_SMALL_INT(depth)));
+}
+
+STATIC void ssl_context_set_verify_callback(mp_obj_ssl_context_t *self, mp_obj_t handler_in) {
+    if (handler_in != mp_const_none && !mp_obj_is_callable(handler_in)) {
+        mp_raise_ValueError(MP_ERROR_TEXT("invalid handler"));
```

Comment:
I don't think it's necessary to validate the handler.  It's an advanced feature and users should pass in a callable.  We can document that and provide examples.

Reply:
I'm not sure what happens when you `mp_call_function` something that is not callable. If that throws an error, sure there is no immediate need to verify.

Reply:
Calls are always verified, so it's not necessary to validate it explicitly.